packages/core/src/encoding.ts

```diff
@@ -35,14 +36,28 @@ export const decodeSilentPaymentAddress = (
 };
 
 export const createLabeledSilentPaymentAddress = (
-    scanKey: Uint8Array,
-    spendKey: Uint8Array,
-    m: Buffer,
+    scanPrivKey: Uint8Array,
+    spendPubKey: Uint8Array,
+    m: number,
     network: Network = bitcoin,
     version: number = 0,
 ) => {
-    spendKey = secp256k1.publicKeyTweakAdd(spendKey, m, true);
-    return encodeSilentPaymentAddress(scanKey, spendKey, network, version);
+    const label = createTaggedHash(
+        'BIP0352/Label',
+        Buffer.concat([scanPrivKey, serialiseUint32(m)]),
+    );
+    const scanPubKey = secp256k1.publicKeyCreate(scanPrivKey);
+    const tweakedSpendPubKey = secp256k1.publicKeyTweakAdd(
+        spendPubKey,
+        label,
+        true,
```

Comment:
Can we add a comment as to what this `true` does here?

Reply:
this is a function provided by `secp256k1` lib. I think we shouldn't need to document the usage of external libs.
To answer your question: it compresses the public key.

Reply:
In a mature library, i would prefer a global constant `const USE_COMPRESSED_PUBKEYS = true` or something and this constant wherever true is passed as that would make things more clear, but this is fine for now

Reply:
I see your point. I'll think about it.
This is done in many places so let's do it in a followup PR.